feat(clp-package)!: Refactor input log config so that S3 source locations are specified in the config file rather than through the CLI. (#788)
y-scope/clp: docs/src/user-guide/guides-using-object-storage/clp-usage.md

## Compressing logs from S3

> To compress logs from S3, use the compress script as follows, replacing the fields in angle brackets
> (`<>`) with the appropriate values:

Reviewer: ```suggestion
To compress logs from S3, use the `sbin/compress.sh` script as follows, replacing the fields in
angle brackets (`<>`) with the appropriate values:
```

```bash
sbin/compress.sh \
  --timestamp-key <timestamp-key> \
  <prefix>
```

> * `<prefix>` is the prefix of all logs you wish to compress and must be relative to the prefix

Author: Yes, this is correct.

> * `<prefix>` is the prefix of all logs you wish to compress and must be relative to the prefix

Reviewer: That seems intuitive as long as our docs are clear.

> * `<prefix>` is the prefix of all logs you wish to compress and must be relative to the prefix

Author: It would also be a simple change to make it so the paths have to be absolute if that's what we want

> * `<prefix>` is the prefix of all logs you wish to compress and must be relative to the prefix

Reviewer 2: It makes sense to use the partial prefix. Just wanted to make sure Kirk is aware since this change is external.

> * `<prefix>` is the prefix of all logs you wish to compress and must be relative to the prefix
>   configured in the [logs-input s3_config][logs-input-s3-config].

Reviewer: ```suggestion
* `<prefix>` is the prefix of all logs you wish to compress and must be relative to
  [logs-input.s3_config.key_prefix][logs-input-s3-config].
  * E.g., if you want to compress the S3 object `/a/b/c.jsonl`, and
    `logs-input.s3_config.key_prefix` is `/a/`, then you would replace `<prefix>` in the command
    above with `b/c.jsonl`.
```

:::{note}
Compressing from S3 only supports a single prefix but will compress any logs that have the given
prefix.

> If you wish to compress a single log file, specify the entire path to the log file. However, if that

Reviewer: Can we add this after the word "file":

```text
(relative to `logs-input.s3_config.key_prefix`)
```

[add-iam-policy]: https://docs.aws.amazon.com/IAM/latest/UserGuide/access_policies_manage-attach-detach.html#embed-inline-policy-console
[aws-region-codes]: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html#Concepts.RegionsAndAvailabilityZones.Availability
[compression-iam-policy]: ./object-storage-config.md#configuration-for-compression

> [logs-input-s3-config]: ./clp-config.md#configuration-for-input-logs

Reviewer: ```suggestion
[logs-input-s3-config]: ./clp-config.md#configuration-for-input-logs

```

We try to end all files with an empty line.
